src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/SecurityTokenHandlerCollection.cs

```diff
@@ -0,0 +1,686 @@
+        /// Gets the List of System.Type of the Token Handlers in this collection.
+        /// </summary>
+        public IEnumerable<Type> TokenTypes
+        {
+            get { return _handlersByType.Keys; }
+        }
+
+        /// <summary>
+        /// Gets the list of Token type Identifier of the Token Handlers.
+        /// </summary>
+        public IEnumerable<string> TokenTypeIdentifiers
+        {
+            get
+            {
+                return _handlersByIdentifier.Keys;
+            }
+        }
+
+        /// <summary>
+        /// Gets a Token Handler by its Token Type Identifier.
+        /// </summary>
+        /// <param name="tokenTypeIdentifier">The Token Type Identfier string to search for.</param>
+        /// <returns>Instance of a SecurityTokenHandler.</returns>
+        public SecurityTokenHandler this[string tokenTypeIdentifier]
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(tokenTypeIdentifier))
+                {
+                    return null;
+                }
+
+                _handlersByIdentifier.TryGetValue(tokenTypeIdentifier, out SecurityTokenHandler handler);
+                return handler;
+            }
+        }
+
+        /// <summary>
+        /// Gets a Token Handler that can handle a given SecurityToken.
+        /// </summary>
+        /// <param name=nameof(token)>SecurityToken for which a Token Handler is requested.</param>
+        /// <returns>Instance of SecurityTokenHandler.</returns>
+        public SecurityTokenHandler this[SecurityToken token]
+        {
+            get
+            {
+                if (null == token)
+                {
+                    return null;
+                }
+
+                return this[token.GetType()];
+            }
+        }
+
+        /// <summary>
+        /// Gets a Token Handler based on the System.Type of the token.
+        /// </summary>
+        /// <param name="tokenType">System.Type of the Token that needs to be handled.</param>
+        /// <returns>Instance of SecurityTokenHandler.</returns>
+        public SecurityTokenHandler this[Type tokenType]
+        {
+            get
+            {
+                SecurityTokenHandler handler = null;
+                if (tokenType != null)
+                {
+                    _handlersByType.TryGetValue(tokenType, out handler);
+                }
+
+                return handler;
+            }
+        }
+
+        /// <summary>
+        /// Creates a system default collection of basic SecurityTokenHandlers, each of which has the system default configuration.
+        /// The SecurityTokenHandlers in this collection must be configured with service specific data before they can be used.
+        /// </summary>
+        /// <returns>A SecurityTokenHandlerCollection with default basic SecurityTokenHandlers.</returns>
+        public static SecurityTokenHandlerCollection CreateDefaultSecurityTokenHandlerCollection()
+        {
```

Comment:
I think this method can't exist as the default set of security token handlers needs to come from DI

Reply:
https://github.com/CoreWCF/CoreWCF/issues/585 => PV2